src/BatteryStateDisplayService.cpp

```diff
@@ -24,10 +24,20 @@ BatteryStateDisplayService::~BatteryStateDisplayService()
 void BatteryStateDisplayService::handleBatteryDataReceived(const BatteryData& batteryData)
 {
     batteryStateOfChargeService_.addData(batteryData);
-
     QTextStream(stdout) << "Voltage: " << batteryData.voltage
-        << " Current: " << batteryData.current
-        << " Total Ah used: " << batteryStateOfChargeService_.totalAmpHoursUsed() << endl;
-
+                        << " Current: " << batteryData.current
+                        << " Total Ah used: " << batteryStateOfChargeService_.totalAmpHoursUsed()<<endl;
     // TODO Print out time till it is depleted or charged.
+    if (batteryStateOfChargeService_.isCharging())
+    {
+        QTextStream(stdout)<<"Time left till full charge: "<<batteryStateOfChargeService_.timeWhenChargedOrDepleted().toString("hh:mm:ss.zzz")<<endl;
```

Comment:
This string constant should be abstracted out to a constant. Using the namespace like you did in other areas. Otherwise, the `<<` are considered operators and so should have spaces around them.